File: src/main/java/com/luneo7/junit4/intellij/state/ListenersConfig.java
package com.luneo7.junit4.intellij.state;

import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.State;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Collections;
import java.util.List;

@State(name = "ListenersConfig")
public class ListenersConfig implements PersistentStateComponent<ListenersConfig.PersistentState> {

    private PersistentState persistentState = new PersistentState();

    @Override
    public @Nullable ListenersConfig.PersistentState getState() {
        return persistentState;
    }

    @Override
    public void loadState(@NotNull PersistentState state) {
        persistentState = state;

    }

    public static class PersistentState {
        private List<String> listeners = Collections.emptyList();

        public final List<String> getListeners() {
            return this.listeners;
        }

        public final void setListeners(List<String> listeners) {
            this.listeners = listeners;
        }
    }
}

File: src/main/kotlin/com/luneo7/junit4/intellij/state/ListenersConfig.kt
package com.luneo7.junit4.intellij.state

import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.State

@State(
    name = "ListenersConfig"
)
open class ListenersConfig : PersistentStateComponent<ListenersConfig.PersistentState> {
    var persistentState: PersistentState = PersistentState()

    override fun getState(): PersistentState {
        return persistentState
    }

    override fun loadState(state: PersistentState) {
        persistentState = state
    }

    class PersistentState {
        var listeners = emptyList<String>()
    }
}